CHANGES.rst

```diff
@@ -16,6 +16,8 @@ New Features
 
 - ``astropy.convolution``
 
+  - Modified cython functions to release the GIL. [#3949]
```

Comment:
This is a pretty nice change and deserves a better changelog entry.  First "cython" -> "Cython", but also might want to note that this enables convolution to be parallelized effectively.